lib.php

```diff
@@ -360,8 +361,10 @@ public function get_links($linkarray) {
             $resubmitlink = ($instructor && $showresubmitlink) ? $this->render_resubmit_link($submission->getid()) : '';
 
             // Output rendered status and resubmission link if applicable.
-            $output .= html_writer::tag('div', $turnitinicon.$status.$resubmitlink,
-                            array('class' => 'turnitinsim_status submission_'.$submissionid));
+            if ($instructor || (!$instructor && $plagiarismsettings->accessstudents)) {
```

Comment:
Is the check if the user is not an instructor necessary? If it is hitting this part of the OR then it will have already evaluated that the user isn't an instructor

Reply:
It also hits this part of the code for the student view too.